### PR TITLE
Change profile endpoint to 127.0.0.1

### PR DESCRIPTION
**Pull request by bitoku**
#### What type of PR is this?

/kind other

#### What this PR does / why we need it:

This PR changes profile endpoint to 127.0.0.1 according to #7448  

#### Which issue(s) this PR fixes:

Fixes a part of #7448

#### Does this PR introduce a user-facing change?

```release-note
Move the tracing profile listener to use 127.0.0.1 as the new default.
```

### CODE REVIEW

File: cmd/crio/main.go [hunk 1]
```diff
@@ -240,7 +240,7 @@ func main() {
 
 		if c.Bool("profile") {
 			profilePort := c.Int("profile-port")
-			profileEndpoint := fmt.Sprintf("localhost:%v", profilePort)
+			profileEndpoint := fmt.Sprintf("127.0.0.1:%v", profilePort)
```
Comment (saschagrunert) [inline cmd/crio/main.go:243]:
```suggestion
			profileEndpoint := fmt.Sprintf("127.0.0.1:%d", profilePort)
```

**Comment by saschagrunert** [comment]
Please also update the completions and docs (see GitHub actions CI).

**Comment by kwilczynski** [comment]
@bitoku @saschagrunert, we allow for the address on which it will listen (bind) to be specified for some other listeners.

An example of this would be the metrics endpoint.

Should we also allow for this to be customised? I cannot see why someone would like to have the profiling endpoint listen on anything other than localhost, but perhaps for the sake of consistency, it should also be done for this endpoint.

Thoughts?

**Comment by saschagrunert** [comment]
> Should we also allow for this to be customised? I cannot see why someone would like to have the profiling endpoint listen on anything other than localhost, but perhaps for the sake of consistency, it should also be done for this endpoint.

Probably, yes. Maybe for testing or something :thinking:

**Comment by bitoku** [comment]
IMHO, we don't have to unless we receive reasonable requests to allow to configure the endpoint.
We've been left this `localhost` since the beginning, and do we get the request?

Making new options will also increase the cost of maintenance (even though it's very little, but every little bit counts).

**Comment by kwilczynski** [comment]
> > Should we also allow for this to be customised? I cannot see why someone would like to have the profiling endpoint listen on anything other than localhost, but perhaps for the sake of consistency, it should also be done for this endpoint.
> 
> Probably, yes. Maybe for testing or something 🤔

@bitoku @saschagrunert same question for the following:

- https://github.com/cri-o/cri-o/pull/8495

**Comment by bitoku** [comment]
Tracing endpoint is already customisable.
I don't think it has to be as well, but because we already have the feature, it's hard to get rid of it.
That's why I feel reluctant to add customisable features which are not that wanted by users.

I'm not strongly opposing because consistency is also important.

**Comment by kwilczynski** [comment]
[...]
> I'm not strongly opposing because consistency is also important.

@bitoku @saschagrunert, one of the goals of the GitHub issue was indeed to make things consistent, if possible.

**Comment by bitoku** [comment]
@kwilczynski 
ok! Do you want me to include that in this PR?
I think it's good to keep them separate to make things small and simple.

**Comment by kwilczynski** [comment]
> @kwilczynski ok! Do you want me to include that in this PR? I think it's good to keep them separate to make things small and simple.


@bitoku @saschagrunert, only if the two of you think it's the right thing to do. We can have a consensus over this.

We do not _have_ to do it. If we feel it's not needed or we are trying to do too much. :smile:

**Comment by bitoku** [comment]
@kwilczynski 
I think we can discuss it separately from this PR. Or do you want to add that change with this change?

**Comment by kwilczynski** [comment]
> @kwilczynski I think we can discuss it separately from this PR. Or do you want to add that change with this change?

@bitoku, sounds good!

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**